### DISCUSSION TITLE
Conversion from docbook: Change reading of the `procedure` element

### DISCUSSION THREAD

**Discussion by tomaskrulis** [jgm/pandoc#9338] [Q&A] [answered]
Hi everyone,

I would like to ask you how I can change the way pandoc reads the docbook element `procedure`: https://tdg.docbook.org/tdg/5.0/procedure 

pandoc seems to read the procedure as a `BulletList` ([permalink to the line in the reader](https://github.com/jgm/pandoc/blob/ff753bdfa62cc335e1c60ca094d3ced027d2f864/src/Text/Pandoc/Readers/DocBook.hs#L936)), and I would like to read it as an `OrderedList`.

How can I go about that? It seems like I might need to create a custom writer. Can I somehow just overload specific parts/functions in the original writer?

Otherwise, I am considering making a custom attribute that would mark the `BulletList`s originating from procedures.

in the past, I have used the Google group which is now inactive it seems. I hope I am using the right place to ask.

**Comment by jgm** [comment, marked as answer]
You'd actually need a custom reader!  Once you get to the writer, the `procedure` element has already been parsed and is indistinguishable from any other BulletList.

I think it would make sense for me to simply change the behavior of the docbook reader, if you wouldn't mind submitting an issue on this.  Please add that we should properly support `substeps` -- it would create a subordinate ordered list.  (It currently just creates subordinate paragraphs.)
